src/main/java/duke/ui/fxcomponents/MainWindow.java

```diff
@@ -0,0 +1,88 @@
+package duke.ui.fxcomponents;
+
+import duke.Duke;
+import duke.ui.GuiHelper;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * MainWindow component of the GUI. Contains additional logic to
+ * format and display messages.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollArea;
+    @FXML
+    private VBox outputContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private GuiHelper guiHelper;
+    private Duke duke;
+
+    private final Image userImage = new Image(this.getClass().getResourceAsStream("/images/user.png"));
+    private final Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/naturo.png"));
+
+    @FXML
+    public void initialize() {
+        scrollArea.vvalueProperty().bind(outputContainer.heightProperty());
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
+        String input = userInput.getText();
+        this.guiHelper.setUserInput(input);
+        this.userMessage(input);
+        this.duke.nextIteration();

```

Comment:
Might want to leave an empty line here for better clarity.

Reply:
Thanks for the review! Glad to know my code helped you. :)

Do you mind elaborating on which classes/methods you think needs more comments/javadocs?